src/fitnesse/testsystems/slim/SlimTestSystem.java

```diff
@@ -118,83 +137,114 @@ protected SlimTestContextImpl createTestContext(TestPage testPage) {
     if (stopTestCalled && !table.isTearDown()) {
       instructionResults = Collections.emptyMap();
+      evaluateTables(assertions, instructionResults);
     } else if (ignoreAllTestsCalled && !table.isTearDown()){
       instructionResults = Collections.emptyMap();
+      evaluateTables(assertions, instructionResults);
     } else {
       boolean tearDownOfAlreadyStartedTest = stopTestCalled && table.isTearDown();
       if(ignoreAllTestsCalled && table.isTearDown()){
         ignoreAllTestsCalled = false;
       }
       if (stopSuiteCalled && !isSuiteTearDownPage && !tearDownOfAlreadyStartedTest) {
         instructionResults = Collections.emptyMap();
+        evaluateTables(assertions, instructionResults);
       } else {
-        instructionResults = slimClient.invokeAndGetResponse(SlimAssertion.getInstructions(assertions));
+        if (isModeSingle()){
+          boolean IgnoreTestTable = false;
+          for (SlimAssertion assertion : assertions) {
+            List<SlimAssertion> oneAssertion = new ArrayList<>();
+            oneAssertion.add(assertion);
+            testAssertionWillBeExecuted(assertion);
+            instructionResults = slimClient.invokeAndGetResponse(SlimAssertion.getInstructions(oneAssertion));
+            final String instructionId = assertion.getInstruction().getId();
+            Object instructionResult = instructionResults.get(instructionId);
+            IgnoreTestTable = evaluateAssertion(instructionResult,IgnoreTestTable, assertion, instructionId);
+          }
+        } else {
+            instructionResults = slimClient.invokeAndGetResponse(SlimAssertion.getInstructions(assertions));
+            evaluateTables(assertions, instructionResults);
+        }
       }
     }
 
-    evaluateTables(assertions, instructionResults);
+    
   }
 
   protected void evaluateTables(List<SlimAssertion> assertions, Map<String, Object> instructionResults) throws SlimCommunicationException {
     boolean IgnoreTestTable = false;
     for (SlimAssertion a : assertions) {
       final String key = a.getInstruction().getId();
-      Object returnValue = instructionResults.get(key);
-
-      //Ignore management
-      if(!ignoreAllTestsCalled) {
-        if (returnValue != null && returnValue.toString().contains(EXCEPTION_IGNORE_ALL_TESTS_TAG)) {
-          ignoreAllTestsCalled = IgnoreTestTable = true;
-        } else if (returnValue != null && returnValue.toString().contains(EXCEPTION_IGNORE_SCRIPT_TEST_TAG)) {
-          IgnoreTestTable = true;
-        } else if (IgnoreTestTable) {
-          returnValue = "IGNORE_SCRIPT_TEST";
-        }
-      } else {
-        returnValue = "IGNORE_SCRIPT_TEST";
+      Object instructionResult = instructionResults.get(key);
+      IgnoreTestTable = evaluateAssertion(instructionResult, IgnoreTestTable, a, key);
+    }
+  }
+
+  private boolean evaluateAssertion(Object InstructionResult, boolean IgnoreTestTable, SlimAssertion a, String key)
```

Comment:
Great improvement to have this in a separate method. I think by making it protected the code using it should be a bit easier to test, and this might aid future changes.


```suggestion
  protected boolean evaluateAssertion(Object instructionResult, boolean ignoreTestTable, SlimAssertion assertion, String key)
```